src/libsystemd/sd-daemon/sd-daemon.c

```diff
@@ -677,11 +681,15 @@ _public_ int sd_notifyf(int unset_environment, const char *format, ...) {
                 r = vasprintf(&p, format, ap);
                 va_end(ap);
 
-                if (r < 0 || !p)
-                        return -ENOMEM;
+                if (r < 0 || !p) {
+                        r = -ENOMEM;
+                        p = mfree(p);  /* If vasprintf failed, do not use the string,
+                                        * even if something was returned. */
+                }
         }
 
-        return sd_pid_notify(0, unset_environment, p);
+        k = sd_pid_notify(0, unset_environment, p);
```

Comment:
RET_GATHER (and everywhere else)?

Reply:
I thought about it, but `RET_GATHER` returns the accumulator, not the newly-added error, so we'd need a temporary variable anyway. And then with just two errors, it's shorter and clearer not to use `RET_GATHER()`.